src/list_filter_profiles.cpp

```diff
@@ -0,0 +1,137 @@
+#include <ros/ros.h>
+#include <xscontroller/xscontrol_def.h>
+#include <xscontroller/xsdevice_def.h>
+#include <xscontroller/xsscanner.h>
+#include <xstypes/xsfilterprofilearray.h>
+
+Journaller * gJournal = 0;
+
+class XdaInterface
+{
+public:
+  XdaInterface()
+  {
+    ROS_INFO("Creating XsControl object...");
+    m_control = XsControl::construct();
+    assert(m_control != 0);
+  }
+  ~XdaInterface()
+  {
+    ROS_INFO("Cleaning up ...");
+    m_control->closePort(m_device);
+    m_control->destruct();
+  }
+
+  bool connectDevice()
+  {
+    // Read baudrate parameter if set
+    XsBaudRate baudrate = XBR_Invalid;
+    if (ros::param::has("~baudrate")) {
```

Comment:
Rosparam has a read option with default. Why not use that instead of implementing it yourself?

Reply:
This is literally copied from `https://github.com/nobleo/xsens_mti_driver/blob/master/src/xdainterface.cpp`. I didn't want to make big changes there to prevent merge conflicts.

Do I then need to cleanup the copied code?

Reply:
Ah, no preferably not.